content/en/blog/general/why-chainsaw-is-unique/index.md

```diff
@@ -0,0 +1,200 @@
+
+## Basic assertions
+
+Let's start by reviewing the basics briefly.
+
+### The most basic assertion
+
+We will start with a basic assertion:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: coredns
+  namespace: kube-system
+spec:
+  replicas: 2
+```
+
+When asking Chainsaw to execute the assertion above, it will look for a Deployment named `coredns` in the `kube-system` namespace and will compare the existing resource with the (partial) resource definition contained in the assertion.
+
+In this specific case, if `spec.replicas` is set to 2 in the existing resource, the assertion will be considered valid, if it is not equal to 2 the assertion will be considered failed.
+
+This is the most basic assertion Chainsaw can evaluate.
+
+### A slightly less basic assertion
+
+A slightly less basic assertion is shown below.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    k8s-app: kube-dns
+  namespace: kube-system
+spec:
+  replicas: 2
+```
+
+This time we are not providing a resource name.
+
+Chainsaw will look up all Deployments with the `k8s-app: kube-dns` label in the `kube-system` namespace. The assertion will be considered valid if **at least one** Deployment matches with the (partial) resource definition contained in the assertion, if none matches the assertion will be considered failed.
+
+Apart from the resource lookup process being a little bit more interesting, this kind of assertion is essentially the same as the previous one. Chainsaw is basically taking a decision by comparing an actual and expected resource.
+
+### A word on timeout
+
+Keep in mind that Chainsaw won't consider an assertion valid or failed in one shot. The assertion will be continuously evaluated during a predetermined period of time until the assertion succeeds or the timeout expires.
+
+## It's good but it's not enough
+
+Now we covered the basic assertions we can look at their limitations and how Chainsaw solves them.
+
+Basic assertions cover the simple cases but:
+
+- Working with arrays is not easy
+    - How are we supposed to compare arrays with different sizes?
+    - How can we check a specific item in an array is present?
+- Matching exact values is not always what we want
+    - How can we verify the number of replicas is above a certain number (not exactly this number)?
+    - How can we apply a regex to a label?
+
+The examples above are the most obvious ones, still they demonstrate that an assertion model needs to be more rich and more flexible than simple comparisons.
+
+### Overview of the Chainsaw Assertion Model
+
+Let's see how Chainsaw can assert that the number of replicas of a Deployment is greater than a certain threshold, let's look at how it can be written.
```

Comment:
```suggestion
Let's see how Chainsaw can assert that the number of replicas of a Deployment is greater than a certain threshold. Let's look at how it can be written.
```

run-on

Reply:
`Let's` appears twice in this short sentence.